sos/plugins/openstack_manila.py

```diff
@@ -21,6 +21,27 @@ class OpenStackManila(Plugin):
     var_puppet_gen = "/var/lib/config-data/puppet-generated/manila"
 
     def setup(self):
+        manila_config_opt = "--config-dir %s/etc/manila/"
+
+        # check eventlet-based service on the baremetal
+        in_ps = self.check_process_by_name("manila-api")
+
+        # check if manila-api is running inside a container
+        # (doesn't matter if running via httpd or as a eventlet based service)
+        in_container = self.running_in_container()
+
+        if in_container:
+            manila_config = manila_config_opt % self.var_puppet_gen
+        else:
+            manila_config = manila_config_opt % ''
+
+        # gather DB version
+        if in_ps or in_container:
+            self.add_cmd_output(
+                "manila-manage " + manila_config + " db version",
+                suggest_filename="manila_db_version"
+            )
```

Comment:
I think this could be condensed a bit, like:

```python
manila_dir = "%s/etc/manila" % (
    self.var_puppet_gen if self.running_in_container() else ''
)
manila_cmd = "manila-manage --config-dir %s db version" % (manila_dir)
self.add_cmd_output(manila_cmd, suggest_filename="manila_db_version")
```

`add_cmd_output` already handles command failures and non-existing commands, so we shouldn't need to have the conditional in place on running the command, just formatting it. Or the `in_ps` check since we can be reasonably sure that `manila_api` is running on the host if this plugin is enabled already.

Reply:
yep, thanks for the clarification. agree this is cleaner.